### PR TITLE
register as a company

### PR DESCRIPTION
**Pull request by BrustureanElena**

### CODE REVIEW

File: src/main/java/com/sintern/api/CompanyController.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    private CompanyService companyService;
+
+    @Autowired
+    public CompanyController(CompanyService companyService) {
+        this.companyService = companyService;
+    }
+
+    @PostMapping("/Register")
+    public void processRegister(@RequestBody Company company) {
+
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/api/CompanyController.java]: Please remove redundant white spaces.

File: src/main/java/com/sintern/service/CompanyService.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+
+    public void addCompany(Company company){
+        String encodedPassword=encodePassword(company.getPassword());
+        company.setPassword(encodedPassword);
+        companyRepository.save(company);
+    }
+
+    public String encodePassword(String password){
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/service/CompanyService.java]: Please move this method in a new Service (e.g. EncrpytionService).

File: src/main/resources/util/create-scripts.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
+INSERT INTO public.student (id, email, password, date_of_birth, first_name, last_name, phone_number) VALUES ('eed75828-3ed7-11ec-9bbc-0242ac130002', 'john.doe@gmail.com', 'johndoe', '1999-01-01', 'John', 'Doe', '0777444333');
```
Comment (DascaluCosmin) [inline src/main/resources/util/create-scripts.sql]: This script file in not in the scope of this Story and should be deleted.

File: src/main/java/com/sintern/service/CompanyService.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import com.sintern.domain.Company;
+import com.sintern.domain.OpenInternPosition;
+import com.sintern.repository.CompanyRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+@Service
+public class CompanyService {
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/service/CompanyService.java]: This should be an interface and have a new class, called `CompanyServiceImpl` that implements the `addCompany(Company)` method

File: src/main/java/com/sintern/api/CompanyController.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import com.sintern.domain.Company;
+import com.sintern.service.CompanyService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/API/Company")
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/api/CompanyController.java]: Please change the endpoint. It should de be `@RequestMapping("/companies")`

File: src/main/java/com/sintern/api/CompanyController.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+public class CompanyController {
+    private CompanyService companyService;
+
+    @Autowired
+    public CompanyController(CompanyService companyService) {
+        this.companyService = companyService;
+    }
+
+    @PostMapping("/Register")
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/api/CompanyController.java]: Please use lowercase letters to conform to the REST standard naming guidelines.

File: src/main/java/com/sintern/service/CompanyServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+import com.sintern.domain.Company;
+import com.sintern.repository.CompanyRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CompanyServiceImpl implements CompanyService {
+    private CompanyRepository companyRepository;
+    private EncryptionService encryptionService= new EncryptionServiceImpl();
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/service/CompanyServiceImpl.java:12]: We use Spring for dependency injection. This is similar to the way we inject a Repository to a Service.

File: src/main/java/com/sintern/service/EncryptionServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+@Service
+public class EncryptionServiceImpl implements EncryptionService {
+
+    public EncryptionServiceImpl() {
+    }
+
+    public String encodePassword(String password){
+        BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+        return passwordEncoder.encode(password);
+
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/service/EncryptionServiceImpl.java]: Does this improve readability? Delete redundant white spaces.

File: src/main/resources/application.yaml [hunk 9]
```diff
@@ -4,7 +4,7 @@ spring:
   datasource:
     url: jdbc:postgresql://localhost:5432/postgres
     username: postgres
-    password: 123
```
Comment (DascaluCosmin) [inline src/main/resources/application.yaml]: Do not expose your database connection credentials. Leave the password and email empty.

File: src/main/java/com/sintern/service/CompanyServiceImpl.java (hunk 7 above)
Comment (BrustureanElena) [reply src/main/java/com/sintern/service/CompanyServiceImpl.java:12]: Done

File: src/main/java/com/sintern/service/EncryptionServiceImpl.java (hunk 8 above)
Comment (BrustureanElena) [reply src/main/java/com/sintern/service/EncryptionServiceImpl.java]: Done

File: src/main/resources/application.yaml (hunk 9 above)
Comment (BrustureanElena) [reply src/main/resources/application.yaml]: Done

**Pull request merged**